### PR TITLE
DAOS-1441 doc: document for DTX

### PR DESCRIPTION
**Pull request by Nasf-Fan**
Describe the DTX basic idea and DTX leader election.

Signed-off-by: Fan Yong <fan.yong@intel.com>
Change-Id: I8b60ae4f29240b6a076110e6e83ea2e6a396cec2

### CODE REVIEW

File: src/vos/README.md [hunk 1]
```diff
@@ -962,3 +964,67 @@ And once the different levels of the trees have been initialized, at no point wi
...
+
+2. If some non-leader replica hits 'prepared' DTX, it will notify the client to retry related read-only RPC with leader replcia.
+Then the leader replica will handle the read-only RPCs that touch uncommitted DTXs.
+
+So genearally, the server on which the leader replica resides will have more load than non-leaders.
+To avoid or reduce the unbalaced load caused by DTX leader, we need multiple leaders on different servers for modifying different objects or different dkeys in the same object. Some rules for the leader election:
+
+R1: If multiple objects share the same redundancy group, then it is better to make the leaders for different objects to be on different servers.
+
+R2: If a multiple replicated object has multiple redundancy groups, then the leaders in different redundancy groups should be on different servers.
```
Comment (jolivier23) [inline src/vos/README.md]: I'm not sure I fully grok what this is trying to say.   Is it referring to different DKEYs ?  Here is a stab

When a replicated object with multiple DKEYs spans multiple redundancy groups, the leaders in different redundancy groups should be on different servers.

File: src/vos/README.md [hunk 2]
```diff
@@ -962,3 +964,67 @@ And once the different levels of the trees have been initialized, at no point wi
...
+
+R3: In each redundancy group, elect the replica which server has the lowest failure sequence as the leader to avoid leader switch as much as possible.
```
Comment (jolivier23) [inline src/vos/README.md]: Servers that fail frequently should be avoided in leader selection to avoid frequent leader migration.

**Comment by jolivier23** [comment]
Please open the actual diff...For some reason, github only selects a few comments on main page.

**Comment by Nasf-Fan** [comment]
@jolivier23 , since you are English native, I agree with your suggestion:

1) Either land the patch firstly, then you can re-org the words when you update vos/README.md, or
2) If you want, you can directly edit current patch and re-push the patch by yourself.

**Comment by jolivier23** [comment]
I'd prefer landing, I think.   @gnailzenh can you land and I will reword tomorrow

**Pull request merged**